### PR TITLE
#414 new entity college

### PR DESCRIPTION
**Pull request by Satendra-SR**
PR for #414 

added new model as College and its migration as
`php artisan make:model College -m`
-new columns to the table
 *college_id
 *college-name
 *address

Director and TPO is not included.
please let me know how I should append these information to the above.

### CODE REVIEW

File: database/migrations/2018_06_14_133520_create_colleges_table.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('colleges', function (Blueprint $table) {
+            $table->increments('id');
+            $table->integer('college_id')->unsigned();
```
Comment (bohraji) [inline database/migrations/2018_06_14_133520_create_colleges_table.php]: The `id` in the college table is the `College ID`. Please remove this table.

File: app/College.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App;
```
Comment (bohraji) [inline app/College.php]: Please move the model inside the `App\Model\HR` namespace.

File: database/migrations/2018_06_14_133520_create_colleges_table.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class CreateCollegesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('colleges', function (Blueprint $table) {
```
Comment (bohraji) [inline database/migrations/2018_06_14_133520_create_colleges_table.php]: Table name for HR modules start with `hr_` Please update.

**Comment by bohraji** [comment]
@rathorevaibhav Please elaborate on how to store information regarding college contacts 'director' and `TPO` info.

**Comment by bohraji** [comment]
Checked this PR :

- [ ] For duplication.
- [ ] For performance. 
- [ ] Usage of recursive methods, loops and algorithms.
- [ ] For readability and understanding. 
- [ ] For functions with multiple responsibilities.
- [ ] For scope of refactoring. 
- [ ] For coupling
- [ ] For alternative approaches to the implementation. 
- [ ] For how the change impacts other parts of the system.
- [ ] For validations. client side and server side validations are present.

For more insights on code reviews checkout our [guidelines](https://docs.google.com/document/d/1Sa-XYJEmGwrOM5gk5zoan_Z6j90ylibmCmt331361nw/edit#).

**Comment by Satendra-SR** [comment]
done with feedback
please review it
@bohraji 
@nishanth-kd

**Comment by rathorevaibhav** [comment]
@bohraji @Satendra-SR sorry for the delay on TPO/Director information. I've added comments here.
https://github.com/ColoredCow/employee-portal/issues/414#issuecomment-397796545

**Comment by Satendra-SR** [comment]
@nishanth-kd 
    *college migration is updated
    *tpo_email is added to the migration

**Comment by nishanth-kd** [comment]
Quite basic

- [ ] For duplication.
- [ ] For performance. 
- [ ] Usage of recursive methods, loops and algorithms.
- [ ] For readability and understanding. 
- [ ] For functions with multiple responsibilities.
- [ ] For scope of refactoring. 
- [ ] For coupling
- [ ] For alternative approaches to the implementation. 
- [ ] For how the change impacts other parts of the system.
- [ ] For validations. client side and server side validations are present.

For more insights on code reviews checkout our [guidelines](https://docs.google.com/document/d/1Sa-XYJEmGwrOM5gk5zoan_Z6j90ylibmCmt331361nw/edit#).

**Pull request merged**